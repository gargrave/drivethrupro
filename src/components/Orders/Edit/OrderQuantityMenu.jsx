import React from 'react'
import { func, number } from 'prop-types'
import { range } from 'lodash'

const OrderQuantityMenu = props => {
  return (
    <span className="input-group">
      <label htmlFor="quantity">Qty: </label>
      <select
        name="quantity"
        id="quantity"
        className="order-quantity-menu"
        value={props.quantity}
        onChange={props.handleQuantityChange}
      >
        {range(1, props.maxRange + 1).map(n =>
          <option key={n} value={n}>
            {n}
          </option>
        )}
      </select>
    </span>
  )
}

OrderQuantityMenu.defaultProps = {
  maxRange: 20
}

OrderQuantityMenu.propTypes = {
  maxRange: number.isRequired,
  quantity: number.isRequired,
  handleQuantityChange: func.isRequired
}

export default OrderQuantityMenu
