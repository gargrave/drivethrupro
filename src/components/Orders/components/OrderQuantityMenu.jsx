import React from 'react'
import { func, number } from 'prop-types'
import { range } from 'lodash'

const OrderQuantityMenu = props => {
  return (
    <span className="input-group">
      <label htmlFor="quantity">Qty: </label>
      <select name="quantity" id="quantity" value={props.quantity} onChange={props.handleQuantityChange}>
        {range(1, 11).map(n =>
          <option key={n} value={n}>
            {n}
          </option>
        )}
      </select>
    </span>
  )
}

OrderQuantityMenu.propTypes = {
  quantity: number.isRequired,
  handleQuantityChange: func.isRequired
}

export default OrderQuantityMenu
