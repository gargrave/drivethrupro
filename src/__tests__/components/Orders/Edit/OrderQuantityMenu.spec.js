import React from 'react'
import { shallow } from 'enzyme'

import OrderQuantityMenu from '../../../../components/Orders/Edit/OrderQuantityMenu'

describe('OrderQuantityMenu', () => {
  let component
  let props

  beforeEach(() => {
    props = {
      quantity: 2,
      handleQuantityChange: jest.fn()
    }
    component = shallow(<OrderQuantityMenu {...props} />)
  })

  test('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  test('pre-selects the option passed into props', () => {
    const select = component.find('.order-quantity-menu')
    const expected = props.quantity
    const actual = select.props().value
    expect(actual).toEqual(expected)
  })

  test('renders a range of 20 if no "maxRange" prop is supplied', () => {
    const actual = component.find('.order-quantity-menu option').length
    const expected = 20
    expect(actual).toEqual(expected)
  })

  test('renders a specific range if a "maxRange" prop is supplied', () => {
    const maxRange = 5
    // re-mount the component with updated props for this test
    props.maxRange = maxRange
    component = shallow(<OrderQuantityMenu {...props} />)
    const actual = component.find('.order-quantity-menu option').length
    expect(actual).toEqual(maxRange)
  })

  test('calls the "change" callback when an option is selected', () => {
    expect(props.handleQuantityChange.mock.calls.length).toEqual(0)
    component.find('.order-quantity-menu').simulate('change')
    expect(props.handleQuantityChange.mock.calls.length).toEqual(1)
  })
})
