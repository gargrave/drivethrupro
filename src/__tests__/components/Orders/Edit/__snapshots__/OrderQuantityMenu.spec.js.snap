// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderQuantityMenu renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <span
    className="input-group"
>
    <label
        htmlFor="quantity"
    >
        Qty: 
    </label>
    <select
        className="order-quantity-menu"
        id="quantity"
        name="quantity"
        onChange={[Function]}
        value={2}
    >
        <option
            value={1}
        >
            1
        </option>
        <option
            value={2}
        >
            2
        </option>
        <option
            value={3}
        >
            3
        </option>
        <option
            value={4}
        >
            4
        </option>
        <option
            value={5}
        >
            5
        </option>
        <option
            value={6}
        >
            6
        </option>
        <option
            value={7}
        >
            7
        </option>
        <option
            value={8}
        >
            8
        </option>
        <option
            value={9}
        >
            9
        </option>
        <option
            value={10}
        >
            10
        </option>
        <option
            value={11}
        >
            11
        </option>
        <option
            value={12}
        >
            12
        </option>
        <option
            value={13}
        >
            13
        </option>
        <option
            value={14}
        >
            14
        </option>
        <option
            value={15}
        >
            15
        </option>
        <option
            value={16}
        >
            16
        </option>
        <option
            value={17}
        >
            17
        </option>
        <option
            value={18}
        >
            18
        </option>
        <option
            value={19}
        >
            19
        </option>
        <option
            value={20}
        >
            20
        </option>
    </select>
</span>,
  "nodes": Array [
    <span
      className="input-group"
>
      <label
            htmlFor="quantity"
      >
            Qty: 
      </label>
      <select
            className="order-quantity-menu"
            id="quantity"
            name="quantity"
            onChange={[Function]}
            value={2}
      >
            <option
                  value={1}
            >
                  1
            </option>
            <option
                  value={2}
            >
                  2
            </option>
            <option
                  value={3}
            >
                  3
            </option>
            <option
                  value={4}
            >
                  4
            </option>
            <option
                  value={5}
            >
                  5
            </option>
            <option
                  value={6}
            >
                  6
            </option>
            <option
                  value={7}
            >
                  7
            </option>
            <option
                  value={8}
            >
                  8
            </option>
            <option
                  value={9}
            >
                  9
            </option>
            <option
                  value={10}
            >
                  10
            </option>
            <option
                  value={11}
            >
                  11
            </option>
            <option
                  value={12}
            >
                  12
            </option>
            <option
                  value={13}
            >
                  13
            </option>
            <option
                  value={14}
            >
                  14
            </option>
            <option
                  value={15}
            >
                  15
            </option>
            <option
                  value={16}
            >
                  16
            </option>
            <option
                  value={17}
            >
                  17
            </option>
            <option
                  value={18}
            >
                  18
            </option>
            <option
                  value={19}
            >
                  19
            </option>
            <option
                  value={20}
            >
                  20
            </option>
      </select>
</span>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <OrderQuantityMenu
        handleQuantityChange={[Function]}
        maxRange={20}
        quantity={2}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "handleQuantityChange": [Function],
          "maxRange": 20,
          "quantity": 2,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <span
          className="input-group"
>
          <label
                    htmlFor="quantity"
          >
                    Qty: 
          </label>
          <select
                    className="order-quantity-menu"
                    id="quantity"
                    name="quantity"
                    onChange={[Function]}
                    value={2}
          >
                    <option
                              value={1}
                    >
                              1
                    </option>
                    <option
                              value={2}
                    >
                              2
                    </option>
                    <option
                              value={3}
                    >
                              3
                    </option>
                    <option
                              value={4}
                    >
                              4
                    </option>
                    <option
                              value={5}
                    >
                              5
                    </option>
                    <option
                              value={6}
                    >
                              6
                    </option>
                    <option
                              value={7}
                    >
                              7
                    </option>
                    <option
                              value={8}
                    >
                              8
                    </option>
                    <option
                              value={9}
                    >
                              9
                    </option>
                    <option
                              value={10}
                    >
                              10
                    </option>
                    <option
                              value={11}
                    >
                              11
                    </option>
                    <option
                              value={12}
                    >
                              12
                    </option>
                    <option
                              value={13}
                    >
                              13
                    </option>
                    <option
                              value={14}
                    >
                              14
                    </option>
                    <option
                              value={15}
                    >
                              15
                    </option>
                    <option
                              value={16}
                    >
                              16
                    </option>
                    <option
                              value={17}
                    >
                              17
                    </option>
                    <option
                              value={18}
                    >
                              18
                    </option>
                    <option
                              value={19}
                    >
                              19
                    </option>
                    <option
                              value={20}
                    >
                              20
                    </option>
          </select>
</span>,
        "_debugID": 2,
        "_renderedOutput": <span
          className="input-group"
>
          <label
                    htmlFor="quantity"
          >
                    Qty: 
          </label>
          <select
                    className="order-quantity-menu"
                    id="quantity"
                    name="quantity"
                    onChange={[Function]}
                    value={2}
          >
                    <option
                              value={1}
                    >
                              1
                    </option>
                    <option
                              value={2}
                    >
                              2
                    </option>
                    <option
                              value={3}
                    >
                              3
                    </option>
                    <option
                              value={4}
                    >
                              4
                    </option>
                    <option
                              value={5}
                    >
                              5
                    </option>
                    <option
                              value={6}
                    >
                              6
                    </option>
                    <option
                              value={7}
                    >
                              7
                    </option>
                    <option
                              value={8}
                    >
                              8
                    </option>
                    <option
                              value={9}
                    >
                              9
                    </option>
                    <option
                              value={10}
                    >
                              10
                    </option>
                    <option
                              value={11}
                    >
                              11
                    </option>
                    <option
                              value={12}
                    >
                              12
                    </option>
                    <option
                              value={13}
                    >
                              13
                    </option>
                    <option
                              value={14}
                    >
                              14
                    </option>
                    <option
                              value={15}
                    >
                              15
                    </option>
                    <option
                              value={16}
                    >
                              16
                    </option>
                    <option
                              value={17}
                    >
                              17
                    </option>
                    <option
                              value={18}
                    >
                              18
                    </option>
                    <option
                              value={19}
                    >
                              19
                    </option>
                    <option
                              value={20}
                    >
                              20
                    </option>
          </select>
</span>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <OrderQuantityMenu
    handleQuantityChange={[Function]}
    maxRange={20}
    quantity={2}
/>,
}
`;
